x-pack/plugins/data_enhanced/public/search/sessions_mgmt/components/table/table.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButton, EuiInMemoryTable, EuiSearchBarProps } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import moment from 'moment';
+import React, { useEffect, useState } from 'react';
+import * as Rx from 'rxjs';
+import { catchError, switchMap } from 'rxjs/operators';
+import { TableText } from '../';
+import { SessionsMgmtConfigSchema } from '../..';
+import { ActionComplete, UISession } from '../../../../../common/search/sessions_mgmt';
+import { SearchSessionsMgmtAPI } from '../../lib/api';
+import { getColumns } from '../../lib/get_columns';
+import { getAppFilter } from './app_filter';
+import { getStatusFilter } from './status_filter';
+
+const TABLE_ID = 'backgroundSessionsMgmtTable';
+
+interface Props {
+  api: SearchSessionsMgmtAPI;
+  initialTable: UISession[] | null;
+  timezone: string;
+  config: SessionsMgmtConfigSchema;
+}
+
+export function SearchSessionsMgmtTable({ api, timezone, initialTable, config, ...props }: Props) {
+  const [tableData, setTableData] = useState<UISession[]>(initialTable ? initialTable : []);
+  const [isLoading, setIsLoading] = useState(false);
+  const [pagination, setPagination] = useState({ pageIndex: 0 });
+
+  // refresh behavior
+  const doRefresh = async () => {
+    setIsLoading(true);
+    await api.fetchTableData().then((results) => {
+      if (results) {
+        setTableData(results);
+      }
+    });
+    setIsLoading(false);
```

Comment:
I think `setIsLoading` is not set to `false` when `fetchTableData` fails. 

Also mixing .then and await seems a bit off here, I'd suggest wrapping in a try/finally